web-client/src/components/CreateKubeconfigButton.js

```diff
@@ -2,23 +2,56 @@ import React, { useState, useEffect } from 'react'
 import axios from 'axios'
 import { Dialog } from '@reach/dialog'
 import Editor from 'react-simple-code-editor'
+import {useRbac} from "../hooks/useRbac";
+import {extractUsersRoles} from "../services/role";
+
+// we extract the valid kubeconfig namespace values
+function getValidNamespaces(roleBindings, clusterRoleBindings, user) {
+    const {extractedPairItems} = extractUsersRoles(roleBindings, clusterRoleBindings, user.name);
+
+    const uniqueNamespaces = extractedPairItems.length === 0 ? [] : [...new Set(extractedPairItems.map(i => i.namespaces).flat(1))];
+
+    // we remove the invalid namespaces from the array
+    const validNamespaces = uniqueNamespaces.filter(i => i !== "ALL_NAMESPACES");
+
+    //a) If no elements are present we add the default namespace to the extracted namespaces. This is done to ensure backwards compatibility
+    //   in the application logic and avoid crashes in the case no namespace is present.
+    //   Default namespace ___SHOULD___ be already used implicitly by kubeconfig if the namespace is not defined
+    //   (that would mean it isn't a breaking change)
```

Comment:
I think it should be removed the info regarding the PR, so after the merge, we have cohesive inline documentation.

Reply:
done 👍